changelog/922.improvement.md

```diff
@@ -0,0 +1,31 @@
+## Problem
+Rasa knowledge base actions cannot infer the object type of an object directly from the user message. This prevents action_query_knowledge from providing a suitable response when a user asks for a certain attribute of an object unless the user first asks to list objects related to that object type even though the knowledge base has the relevant information. That is, the knowledge base actions require the slot `object_type` to be set to one of the primary key values in the knowledge base for it to search through the objects. Here is an example:
+```
+Your input ->  what is the price range of Berlin Burrito Company?
+Sorry, I'm not sure I understand. Can you rephrase?
+Your input ->  list some restaurants
+Found the following objects of type 'restaurant':
+1: Gong Gan
+2: I due forni
+3: Pfefferberg
+4: Lụa Restaurant
+5: Donath
+Your input ->  what is the price range of Berlin Burrito Company?
+'Berlin Burrito Company' has the value 'cheap' for attribute 'price-range'.
+```
+
+## Proposed solution
+- The improvement requires changes to the classes ActionQueryKnowledgeBase and InMemoryKnowledgeBase under rasa-sdk. 
+- The `object_type` can be inferred by  utilizing the entity classification (DIET) where object types are used as entities to annotate object names. 
+This also requires changes to be made to slot management to enable dynamic inference of `object_type`. 
+- The scope of suggested solution is limited to user queries where they ask for an attribute of a given object without mentioning the object type and without having the need to first ask a list of options of corresponding object type. 
+- E.g: If the user asks ‘price range of Berlin Burrito Company’, then rasa will extract and set attribute slot value to ‘price-range’ and hotel slot value to ‘Berlin Burrito Company’. From this it can be inferred that the user is talking about object type ‘hotel’. 
```

Comment:
Instead of saying "rasa will extract..." it would be better to say the "knowledge base query function will extract..."